### PR TITLE
Add Flow annotations for ReactInputSelection.js

### PR DESCRIPTION
**Pull request by cmatheson**
Hi,

I added some type checks to this relatively easy-looking area of React.

Thanks!

### CODE REVIEW

File: src/renderers/dom/client/ReactInputSelection.js [hunk 1]
```diff
@@ -29,13 +37,10 @@ function isInDocument(node) {
...
 
-  hasSelectionCapabilities: function(elem) {
-    var nodeName = elem && elem.nodeName && elem.nodeName.toLowerCase();
-    return nodeName && (
-      (nodeName === 'input' && elem.type === 'text') ||
-      nodeName === 'textarea' ||
-      elem.contentEditable === 'true'
-    );
+  hasSelectionCapabilities: function(elem : HTMLElement) {
+    return (elem instanceof HTMLInputElement && elem.type === 'text') ||
```
Comment (Aweary) [inline src/renderers/dom/client/ReactInputSelection.js]: Why did you change this logic here? We shouldn't change implementation logic unless there's a good reason to do it. We also don't want to include changes like this alongside stylistic changes like flowtype annotations, as it would make it hard to revert just this change if we needed to.

**Comment by cmatheson** [comment]
I'm not getting that type error in travis locally... Not sure why document would have an unknown type.  I'll try to reproduce.

**Comment by gaearon** [comment]
Closing as stale. If somebody wants to pick it up please feel free to!
Thanks for the PR.
